disk_usage/disk_usage.py

```diff
@@ -27,40 +28,33 @@
 
 
 def aggregate_level(name: str) -> str:
-    """Returns a prefix for the given blob name at the aggregation level."""
+    """Returns a prefix for the given blob name at the folder or Hail table level."""
     ht_index = name.find('.ht/')
     if ht_index != -1:
         return name[: ht_index + 3]
     mt_index = name.find('.mt/')
     if mt_index != -1:
         return name[: mt_index + 3]
-    slash_index = name.find('/')
+    slash_index = name.rfind('/')
     if slash_index == -1:
-        return name
-    next_slash = name.find('/', slash_index + 1)
-    if next_slash != -1:
-        slash_index = next_slash
+        return ''  # Root level
     return name[:slash_index]
 
 
-@dataclass
-class AggregateStats:
-    """Aggregate stats values."""
-
-    size: int = 0
-    num_blobs: int = 0
-
-
 def main():
     """Main entrypoint."""
     # Don't print DEBUG logs from urllib3.connectionpool.
     logging.getLogger().setLevel(logging.INFO)
 
     storage_client = storage.Client()
     dataset = get_config()['workflow']['dataset']
+    access_level = get_config()['workflow']['access_level']
 
-    aggregate_stats = defaultdict(AggregateStats)
+    aggregate_stats = defaultdict(lambda: defaultdict(int))
     for bucket_suffix in BUCKET_SUFFIXES:
+        if access_level == 'test' and not bucket_suffix.startswith('test'):
+            continue  # Skip main buckets when testing.
+
         bucket_name = f'cpg-{dataset}-{bucket_suffix}'
         logging.info(f'Listing blobs in {bucket_name}...')
         blobs = storage_client.list_blobs(bucket_name)
```

Comment:
Does this not already list every single file in the bucket? I thought you had to be explicit about filtering to just the bucket root or something.